src/main/java/com/amazonaws/services/dynamodbv2/AmazonDynamoDBLockClient.java

```diff
@@ -1141,6 +1141,11 @@ public void sendHeartbeat(final SendHeartbeatOptions options) {
                 final long lastUpdateOfLock = LockClientUtils.INSTANCE.millisecondTime();
                 this.dynamoDB.updateItem(updateItemRequest);
                 lockItem.updateRecordVersionNumber(recordVersionNumber, lastUpdateOfLock, leaseDurationToEnsureInMilliseconds);
+                if (deleteData) {
```

Comment:
Please add tests for changes to sendHeartbeat methods in https://github.com/awslabs/amazon-dynamodb-lock-client/blob/5747293316880bd8ab375affefa863a5a1e50ad7/src/test/java/com/amazonaws/services/dynamodbv2/AmazonDynamoDBLockClientTest.java

Reply:
Added in the latest commit.